src/main/kotlin/app/revanced/patcher/Patcher.kt

```diff
@@ -1,184 +1,60 @@
-                            buildOptions.isFramework = metaInfo.isFrameworkApk
-                            buildOptions.resourcesAreCompressed = metaInfo.compressionType
-                            buildOptions.doNotCompress = metaInfo.doNotCompress
-                        }
-
-                        resources.setSdkInfo(metaInfo.sdkInfo)
-                        resources.setVersionInfo(metaInfo.versionInfo)
-                        resources.setSharedLibrary(metaInfo.sharedLibrary)
-                        resources.setSparseResources(metaInfo.sparseResources)
-                    }
-
-                    val manifestFile = cacheDirectory.resolve("AndroidManifest.xml")
-
-                    ResXmlPatcher.fixingPublicAttrsInProviderAttributes(manifestFile)
-
-                    val aaptFile = cacheDirectory.resolve("aapt_temp_file")
-
-                    // delete if it exists
-                    Files.deleteIfExists(aaptFile.toPath())
-
-                    val resDirectory = cacheDirectory.resolve("res")
-                    val includedFiles = metaInfo.usesFramework.ids.map { id ->
-                        androlibResources.getFrameworkApk(
-                            id, metaInfo.usesFramework.tag
-                        )
-                    }.toTypedArray()
-
-                    logger.info("Compiling resources")
-                    androlibResources.aaptPackage(
-                        aaptFile, manifestFile, resDirectory, null, null, includedFiles
-                    )
-
-                    resourceFile = aaptFile
-                } finally {
-                    cacheDirectory.close()
-                }
-            }
-
-            else -> logger.info("Not compiling resources because resource patching is not required")
-        }
-
-        logger.trace("Creating new dex file")
-        val newDexFile = object : DexFile {
-            override fun getClasses() = context.bytecodeContext.classes.also { it.replaceClasses() }
-            override fun getOpcodes() = this@Patcher.opcodes
-        }
-
-        // write modified dex files
-        logger.info("Writing modified dex files")
-        val dexFiles = mutableMapOf<String, MemoryDataStore>()
-        MultiDexIO.writeDexFile(
-            true, -1, // core count
-            dexFiles, NAMER, newDexFile, DexIO.DEFAULT_MAX_DEX_POOL_SIZE, null
-        )
-
-        return PatcherResult(
-            dexFiles.map {
-                app.revanced.patcher.util.dex.DexFile(it.key, it.value.readAt(0))
-            },
-            metaInfo.doNotCompress?.toList(),
-            resourceFile
-        )
-    }
+    fun addIntegrations(integrations: List<File>) = context.integrations.add(integrations)
 
     /**
      * Add [Patch]es to the patcher.
      * @param patches [Patch]es The patches to add.
      */
-    fun addPatches(patches: Iterable<Class<out Patch<Context>>>) {
+    fun addPatches(patches: Iterable<PatchClass>) {
```

Comment:
It might be worth to ditch the "add" methods in favour of passing the patches and integrations in the constructor as parameters. A deconstructor may also be added at some point because there are some object classes used.

Reply:
I have replaced them with constructor parameters.
You can mark this as resolved if you are happy with how I did it.